addon-test-support/index.js

```diff
@@ -10,8 +10,16 @@ export function setOptions(options = {}) {
   errorOnGlobalSinonAccess = !!options.errorOnGlobalSinonAccess;
 }
 
+export function setup() {
+  let currentTest = QUnit.config.current;
+
+  currentTest.module.hooks.beforeEach.push(createSandbox);
+  currentTest.module.hooks.afterEach.push(restoreSandbox);
```

Comment:
Chatted with @scalvert in slack about this.  As written this will slowly add more and more `restoreSandbox` / `createSandbox` entries into the `QUnit.config.current.module`'s list of hooks (once for each test within the module). This is because the module is shared amongst all of its tests.